app/models/user.rb

```diff
@@ -26,6 +26,6 @@ def customer?
   end
 
   def has_restaurant?
-    self.restaurant != nil
+    self.restaurant != nil && self.restaurant.id != nil
```

Comment:
There's one scenario under which `self.restaurant != nil` is not enough: when you are creating your restaurant for the first time and lack a validation thing (name, address). In that case, you "have" a restaurant (it's staged but not saved, and `current_user.restaurant` is not nil). In that one scenario, this method fails in its job of telling us whether the user actually has a restaurant or not.